recipes_source/xeon_run_cpu.rst

```diff
@@ -0,0 +1,291 @@
+Logical cores are indexed afterward. 112-167 are 56 logical cores on the first NUMA socket,
+168-223 are the second 56 logical cores on the second NUMA socket.
+Typically, running PyTorch programs with compute intense workloads should avoid using logical cores to get good performance.
+
+.. code-block:: console
+
+   $ lscpu
+   ...
+   CPU(s):                  224
+     On-line CPU(s) list:   0-223
+   Vendor ID:               GenuineIntel
+     Model name:            Intel (R) Xeon (R) CPU Max 9480
+       CPU family:          6
+       Model:               143
+       Thread(s) per core:  2
+       Core(s) per socket:  56
+       Socket(s):           2
+   ...
+   NUMA:
+     NUMA node(s):          2
+     NUMA node0 CPU(s):     0-55,112-167
+     NUMA node1 CPU(s):     56-111,168-223
+   ...
+
+Linux provides a tool, ``numactl``, that allows user control of NUMA policy for processes or shared memory.
+It runs processes with a specific NUMA scheduling or memory placement policy.
+As described above, cores share high-speed cache in one socket, thus it is a good idea to avoid cross socket computations.
+From a memory access perspective, bounding memory access locally is much faster than accessing remote memories.
+``numactl`` command should have been installed in recent Linux distributions. In case it is missing, we can install it manually with the installation command, like
+
+.. code-block:: console
+
+   $ apt-get install numactl
+
+on Ubuntu, or
+
+.. code-block:: console
+
+   $ yum install numactl
+
+on CentOS.
+
+The ``taskset`` command in Linux is another powerful utility that allows you to set or retrieve the CPU affinity of a running process.
+``taskset`` are pre-installed in most Linux distributions and in case it's not, we can install it with command
+
+.. code-block:: console
+
+   $ apt-get install util-linux
+
+on Ubuntu, or
+
+.. code-block:: console
+
+   $ yum install util-linux
+
+on CentOS.
+
+OpenMP
+~~~~~~
+
+OpenMP is an implementation of multithreading, a method of parallelizing where a primary thread (a series of instructions executed consecutively) forks a specified number of sub-threads and the system divides a task among them. The threads then run concurrently, with the runtime environment allocating threads to different processors.
+Users can control OpenMP behaviors with some environment variable settings to fit for their workloads, the settings are read and executed by OMP libraries. By default, PyTorch uses GNU OpenMP Library (GNU libgomp) for parallel computation. On Intel(R) platforms, Intel(R) OpenMP Runtime Library (libiomp) provides OpenMP API specification support. It usually brings more performance benefits compared to libgomp.
+
+The Intel(R) OpenMP Runtime Library can be installed via the command
+
+.. code-block:: console
+
+   $ pip install intel-openmp
+
+or
+
+.. code-block:: console
+
+   $ conda install mkl
+
+Memory Allocator
```

Comment:
Same as above - I wish the heading could be a bit more descriptive - maybe "Choosing Memory Allocators"?